pydantic/fields.py

```diff
@@ -355,6 +355,18 @@ def _type_analysis(self) -> None:  # noqa: C901 (ignore complexity)
         origin = getattr(self.type_, '__origin__', None)
         if origin is None:
             # field is not "typing" object eg. Union, Dict, List etc.
+            if self.type_ is tuple:
+                self.shape = SHAPE_TUPLE
+                return
+            if self.type_ is list:
+                self.shape = SHAPE_LIST
+                return
+            if self.type_ is set:
+                self.shape = SHAPE_SET
+                return
+            if self.type_ is frozenset:
+                self.shape = SHAPE_FROZENSET
+                return
```

Comment:
1. I think this would be cleaner using `elif` and only a single return (the one that was already there), closer to:
```
if self.type_ is x:
    self.shape = SHAPE_X
elif ...
elif ...
...
return
```

2. @samuelcolvin Should we test for subclasses? (@kousikmitra in case this change is desired/necessary, it should be with `if lenient_issubclass(self.type_, x)` (need to use `lenient_issubclass`, which is a pydantic util function, in order to prevent certain errors).

Reply:
Maybe even
``` py
TYPES_TO_SHAPES = {
    tuple: SHAPE_TUPLE,
    list: SHAPE_LIST,
    set: SHAPE_SET,
    frozenset: SHAPE_FROZENSET,
}
# ...
if origin is None:
    self.shape = TYPES_TO_SHAPE[self.type_]
```

Reply:
Well, should probably use `TYPES_TO_SHAPE.get(self.type_)` and check whether it is `None` in case the value isn't one of those types. But yes, that would be cleaner.

Reply:
Yes, It's looking much better. I am making the changes.

Reply:
I have made the changes. Do I need to update the test cases as well?

Reply:
@kousikmitra do you know why this is failing the checks? Are tests and linting etc passing for you locally? (If you run `make` locally, does it succeed? See https://pydantic-docs.helpmanual.io/contributing/ for more detail about how to run the various CI tests locally if you haven't done that yet.)

Reply:
If there are  any failing tests then yes, you should update them. Also, it would be great if you could add a basic test demonstrating that this actually fixes #1112 (probably need that before it gets merged).

If you don't have time or aren't sure how to do that, I can take care of it at some point soon, just let me know.

Reply:
I have ran `make` locally, There is almost 20 test cases which are failing but other checks like code format, linting etc are passing.  I will look into the test cases and revert you back if I need any help.

Reply:
Hi @dmontagu! I think there is still a thing needs to be addressed. There is also a member `sub_fields` which is getting initialized in case of `typing` module objects. But in case of standard sequence type it's not getting initialized here. Can you just check that?